allennlp/training/trainer.py

```diff
@@ -631,6 +631,9 @@ def train(self) -> None:
 
         logger.info("Beginning training.")
 
+        train_metrics = {} #type: Dict[str, float]
```

Comment:
Instead of commenting with `# type:`, just given the type inline: `train_metrics: Dict[str, float] = {}`.  Same for the line below.

Also, just for future reference, the style we typically use with inline comments like this is at least two spaces before the `#` and one space after.

Reply:
Thanks--I wasn't sure whether that was valid syntax and I found the annotation I used in Google.

On your second note, I'd love if we could programmatically enforce all the style we care about (and ignore the rest).

Reply:
I imagine some aspects of style are impossible to enforce programmatically.  This one seems feasible, though, and it would be great to add to our linter.  Not sure if pylint or any other linter checks for this.